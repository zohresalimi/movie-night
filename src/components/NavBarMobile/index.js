import React, { useState } from "react";
import { Grid, Icon, Menu, Sidebar } from "semantic-ui-react";
import { Link } from "react-router-dom";

export default function NavBarMobile() {
  const [isVisible, setIsVisible] = useState(false);
  const [active, setActive] = useState("search");

  const handleItemClick = (e) => {
    setActive(e.target.innerText.toLowerCase());
    setIsVisible(false);
  };

  return (
    <>
      <Grid columns={1}>
        <Grid.Row>
          <Menu fixed="top" inverted>
            <Menu.Item onClick={() => setIsVisible(!isVisible)}>
              <Icon name="sidebar" />
            </Menu.Item>
          </Menu>
          <Grid.Column>
            <Sidebar
              as={Menu}
              animation="overlay"
              icon="labeled"
              inverted
              onHide={() => setIsVisible(false)}
              vertical
              visible={isVisible}
              width="thin"
            >
              <Menu.Item
                as={Link}
                to="/"
                name="search"
                active={active === "search"}
                onClick={handleItemClick}
              >
                <Icon name="search" />
                Search
              </Menu.Item>
              <Menu.Item
                as={Link}
                to="/watch-later"
                name="watch later"
                data-testid="watch-later"
                active={active === "watch later"}
                onClick={handleItemClick}
              >
                <Icon name="clock" />
                Watch Later
              </Menu.Item>
              <Menu.Item
                as={Link}
                to="/favorites"
                name="favorites"
                active={active === "favorites"}
                onClick={handleItemClick}
              >
                <Icon name="star" />
                Favorites
              </Menu.Item>
            </Sidebar>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </>
  );
}
