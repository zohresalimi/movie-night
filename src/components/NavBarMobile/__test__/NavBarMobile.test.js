import * as React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, fireEvent, act } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import NavBarMobile from "../";

async function renderWrapper() {
  let component;

  act(() => {
    component = render(
      <Router>
        <NavBarMobile />
      </Router>
    );
  });

  return component;
}

describe("NavBarMobile Component Testing", () => {
  test("take snapshot", async () => {
    const { container } = await renderWrapper();
    expect(container).toMatchSnapshot();
  });

  test("match snapshot if active item is changed", async () => {
    const { container, getByTestId } = await renderWrapper();
    const menuItem = getByTestId("watch-later");
    fireEvent.click(menuItem, { target: { innerText: "Watch Later" } });
    expect(container).toMatchSnapshot();
  });
});
