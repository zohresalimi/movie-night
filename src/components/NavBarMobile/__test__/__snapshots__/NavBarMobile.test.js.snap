// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarMobile Component Testing match snapshot if active item is changed 1`] = `
<div>
  <div
    class="ui one column grid"
  >
    <div
      class="row"
    >
      <div
        class="ui inverted top fixed menu"
      >
        <a
          class="item"
        >
          <i
            aria-hidden="true"
            class="sidebar icon"
          />
        </a>
      </div>
      <div
        class="column"
      >
        <div
          class="ui inverted vertical labeled icon ui overlay left thin sidebar menu"
        >
          <a
            class="item"
            href="/"
          >
            <i
              aria-hidden="true"
              class="search icon"
            />
            Search
          </a>
          <a
            class="active item"
            data-testid="watch-later"
            href="/watch-later"
          >
            <i
              aria-hidden="true"
              class="clock icon"
            />
            Watch Later
          </a>
          <a
            class="item"
            href="/favorites"
          >
            <i
              aria-hidden="true"
              class="star icon"
            />
            Favorites
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavBarMobile Component Testing take snapshot 1`] = `
<div>
  <div
    class="ui one column grid"
  >
    <div
      class="row"
    >
      <div
        class="ui inverted top fixed menu"
      >
        <a
          class="item"
        >
          <i
            aria-hidden="true"
            class="sidebar icon"
          />
        </a>
      </div>
      <div
        class="column"
      >
        <div
          class="ui inverted vertical labeled icon ui overlay left thin sidebar menu"
        >
          <a
            class="active item"
            href="/"
          >
            <i
              aria-hidden="true"
              class="search icon"
            />
            Search
          </a>
          <a
            class="item"
            data-testid="watch-later"
            href="/watch-later"
          >
            <i
              aria-hidden="true"
              class="clock icon"
            />
            Watch Later
          </a>
          <a
            class="item"
            href="/favorites"
          >
            <i
              aria-hidden="true"
              class="star icon"
            />
            Favorites
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
